### PR TITLE
moving model logic from runners to models, specifically encoded patterns

### PR DESCRIPTION
**Pull request by BoltzmannBrain**

### CODE REVIEW

**Comment by BoltzmannBrain** [comment]
@scottpurdy please review. The priority of this PR is to implement `queryModel()` methods, which called for moving model-specific logic out of the experiment runners and into the classification models; this has been in my todo list for some time now. There is still much I intend to do in this effort, but will implement the changes (and cleaning up) in future PRs in order to limit the scope here.

File: fluent/models/classify_endpoint.py [hunk 1]
```diff
@@ -36,11 +36,13 @@ class ClassificationModelEndpoint(ClassificationModel):
   From the experiment runner, the methods expect to be fed one sample at a time.
   """
 
-  def __init__(self, verbosity=1, numLabels=3):
+  def __init__(self, verbosity=1, numLabels=3,
+    modelDir="ClassificationModelEndpoint"):
```
Comment (scottpurdy) [inline fluent/models/classify_endpoint.py]: fix indentation

File: fluent/models/classify_endpoint.py [hunk 2]
```diff
@@ -36,11 +36,13 @@ class ClassificationModelEndpoint(ClassificationModel):
...
     """
     Initialize the encoder as CioEncoder; requires a valid API key.
     """
-    super(ClassificationModelEndpoint, self).__init__(verbosity, numLabels)
+    super(ClassificationModelEndpoint, self).__init__(verbosity=verbosity,
+      numLabels=numLabels, modelDir=modelDir)
```
Comment (scottpurdy) [inline fluent/models/classify_endpoint.py]: fix indentation

File: fluent/models/classify_htm.py [hunk 3]
```diff
@@ -33,8 +33,10 @@ class ClassificationModelHTM(ClassificationModel):
   Class to run the survey response classification task with nupic network
   """
 
-  def __init__(self, inputFilePath, verbosity=1, numLabels=3, spTrainingSize=0,
-               tmTrainingSize=0, clsTrainingSize=0, classifierType="KNN"):
+  def __init__(self, inputFilePath, verbosity=1, numLabels=3,
+    modelDir="ClassificationModelHTM",
```
Comment (scottpurdy) [inline fluent/models/classify_htm.py]: fix indentation

**Comment by scottpurdy** [comment]
You should run pylint on the files. There isn't a pylintrc in this repo yet so you will have to manually point to the one in nupic (make sure you have latest nupic)

**Comment by BoltzmannBrain** [comment]
Thanks @scottpurdy I didn't know we have a pylint rcfile, cool! Cleaned it up a bit, take another look.

File: fluent/models/classify_endpoint.py [hunk 4]
```diff
@@ -227,3 +232,13 @@ def getWinningLabels(distances, numLabels, metric):
 
     return numpy.array(
         [distances.keys()[catIdx] for catIdx in sortedIdx[:numLabels]])
+
+
+  @staticmethod
+  def query():
+    print "The Classifciation Endpoint model doesn't support this method."
```
Comment (scottpurdy) [inline fluent/models/classify_endpoint.py]: typo here and below

**Comment by scottpurdy** [comment]
:+1: although functionality wise I don't really know the code well enough

**Comment by BoltzmannBrain** [comment]
Thanks for the review, some of the functionality is going to change in the next couple PRs anyways :)

**Pull request merged**